test/unit/step-value-collection-test.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const assert = require('assert');
+const sinon  = require('sinon');
+
+const StepValue           = require('../../lib/step-value');
+const StepValueCollection = require('../../lib/step-value-collection');
+
+describe('StepValueCollection', function() {
+  let collection, validStep, ambiguousStep, notFoundStep;
+
+  beforeEach(function() {
+    validStep = new StepValue(
+      {text: 'I am the step text', keyword: 'Given'},
+      [{implementation: sinon.spy()}]
+    );
+
+    ambiguousStep = new StepValue(
+      {text: 'I live and breath', keyword: 'When'},
+      [{implementation: sinon.spy()}, {implementation: sinon.spy()}]
+    );
+
+    notFoundStep = new StepValue(
+      {text: 'I will be validated', keyword: 'Then'},
+      []
+    );
+  });
+
+  it('is valid if all the steps are valid', function() {
+    collection = new StepValueCollection([validStep, validStep, validStep]);
+    assert(collection.valid());
+    collection = new StepValueCollection([validStep, ambiguousStep]);
+    assert(!collection.valid());
+  });
+
+  it('returns ambiguous steps', function() {
+    collection = new StepValueCollection([validStep, ambiguousStep, ambiguousStep]);
+    let ambiguousSteps = collection.ambiguousSteps();
+    assert.equal(ambiguousSteps.length, 2);
+    assert.equal(ambiguousSteps[0], ambiguousStep);
```

Comment:
Not sure if this is intended, but this assertion makes sure that the references are equal not the actual struct. As a consumer of `StepValueCollection` do I need to make sure that it keeps the original step references or do I not care if it modifies/copies/etc under the covers? If the ladder, then we should probably change these assertions to `deepEqual`